lib/active_model/serializer/adapter/json.rb

```diff
@@ -37,9 +37,7 @@ def serializable_hash(options = {})
             @result = @core.merge @hash
           end
 
-          if root = options.fetch(:root, serializer.json_key)
-            @result = { root => @result }
-          end
+          @result = { root => @result } if root
           @result
```

Comment:
I think I prefer the latter but not enough to change existing code.  (Though, if I were a lib owner and felt like I owned the style, I might change it.)  

I prefer the latter essentially because of the combination of the conditional modifier at the end of the line and the enveloping mutation of the return value add a bit of mental overhead for me to see what's going on.

I'd probably even consider (and then dismiss) `root ? { root => @result } : @result`

dem's my two cents.